File: src/main/java/com/kishlaly/ta/analyze/tasks/blocks/one/Long_ScreenOne_LastBarGreen.java
package com.kishlaly.ta.analyze.tasks.blocks.one;

import com.kishlaly.ta.model.BlockResultJava;
import com.kishlaly.ta.model.SymbolDataJava;
import com.kishlaly.ta.utils.LogJava;

import static com.kishlaly.ta.model.BlockResultCodeJava.LAST_QUOTE_NOT_GREEN_SCREEN_1;
import static com.kishlaly.ta.model.BlockResultCodeJava.OK;

/**
 * the last quote is green
 */
public class Long_ScreenOne_LastBarGreen implements ScreenOneBlock {
    @Override
    public BlockResultJava check(SymbolDataJava screen) {
        boolean lastBarIsGreen = screen.getLastQuote().getOpen() < screen.getLastQuote().getClose();
        if (!lastBarIsGreen) {
            LogJava.recordCode(LAST_QUOTE_NOT_GREEN_SCREEN_1, screen);
            LogJava.addDebugLine("The last quote is not green on the long-term screen");
            return new BlockResultJava(screen.getLastQuote(), LAST_QUOTE_NOT_GREEN_SCREEN_1);
        }
        return new BlockResultJava(screen.getLastQuote(), OK);
    }
}

File: src/main/kotlin/com/kishlaly/ta/analyze/tasks/blocks/one/Long_ScreenOne_LastBarGreen.kt
package com.kishlaly.ta.analyze.tasks.blocks.one

import com.kishlaly.ta.model.BlockResult
import com.kishlaly.ta.model.BlockResultCode
import com.kishlaly.ta.model.SymbolData
import com.kishlaly.ta.utils.Log

/**
 * the last quote is green
 */
class Long_ScreenOne_LastBarGreen : ScreenOneBlock {
    override fun check(screen: SymbolData): BlockResult {
        val lastBarIsGreen = screen.lastQuote.open < screen.lastQuote.close
        if (!lastBarIsGreen) {
            Log.recordCode(BlockResultCode.LAST_QUOTE_NOT_GREEN_SCREEN_1, screen)
            Log.addDebugLine("The last quote is not green on the long-term screen")
            return BlockResult(screen.lastQuote, BlockResultCode.LAST_QUOTE_NOT_GREEN_SCREEN_1)
        }
        return BlockResult(screen.lastQuote, BlockResultCode.OK)

    }
}